terasoluna-gfw-functionaltest-domain/src/main/java/org/terasoluna/gfw/functionaltest/domain/service/codelist/CodeListServiceImpl.java

```diff
@@ -46,7 +46,7 @@
 
     @Override
     public ItemCode findOne(Integer code) {
-        return codeListRepository.findOne(code);
+        return codeListRepository.findById(code).orElse(null);
```

Comment:
- [ ] Uh... Bad solution.

* CodeListController
```
    @RequestMapping(value = "next", method = RequestMethod.POST, params = "jdbcCodeListTestUpdateCodeList")
    public String testJdbcCodeListTestUpdateCodeList(CodeListMngForm form,
            Model model) {
        ItemCode row = codeListService.findOne(form.getId());
        row.setCode(form.getCode());
        row.setLabel(form.getValue());
        codeListService.save(row);
        return "codelist/jdbcCodeListTestUpdateCodeList";
    }
```

If the service returns `null`, the controller will throw an NPE.
I think taking advantage of the characteristics of `Optional` and taking out the contents only when the value exists using `ifPresent`.

Reply:
[Temporal solution]
Adjust way of modification to guideline functional test.